src/backend/opencl/kernel/iops.cl

```diff
@@ -7,13 +7,20 @@
  * http://arrayfire.com/licenses/BSD-3-Clause
  ********************************************************/
 
+#if CPLX
+    #define IS_NAN(in) !((in.x) == (in.x)) || !((in.y) == (in.y))
+#else
+    #define IS_NAN(in) !((in) == (in))
+#endif
+
 #if CPLX
 #define sabs(in) ((in.x)*(in.x) + (in.y)*(in.y))
 #ifdef MIN_OP
 void binOp(T *lhs, uint *lidx, T rhs, uint ridx)
 {
-    if ((sabs(lhs[0]) > sabs(rhs)) ||
-        (sabs(lhs[0]) == sabs(rhs) && *lidx < ridx)) {
+    if (!(IS_NAN(lhs[0])) && 
```

Comment:
I think if we reverse the order of the && operation we can avoid calling IS_NAN as much because of short-circuiting.

so instead of
```
if (!(IS_NAN(lhs[0])) &&
    ((sabs(lhs[0]) < sabs(rhs)) || (sabs(lhs[0]) == sabs(rhs)))) {
```
```
if ((sabs(lhs[0]) < sabs(rhs)) || (sabs(lhs[0]) == sabs(rhs)))
    && !(IS_NAN(lhs[0])))
```

Reply:
Oh yeah that makes sense - that comparison is probably more often useful than the is_nan check anyway. Btw what do you think of making the is_nan check an inline function than a macro? Just to make it safer from potential un-obvious syntactical errors next time we might have to deal with the is_nan check?

Reply:
Fewer macros the better. If you don't run into any issues then you should do it.